### PR TITLE
Fix bad escaping, comparison and nonexistent getError().

### PR DESCRIPTION
**Pull request by lud2k**
This commit fixes a few issues with the latest version of the
SDK.
- The code was using != to do string comparison. This is not
valid in java. Changed to StringUtils.isNotEmpty.
- The code was calling getError() which doesn't exist. It was
removed a while back in a previous commit.
- The code that is trying to fix escaping issues for the
signature is causing issues when given a string as requestPacket.
Disabled fixing escaping if it's already a string.

### CODE REVIEW

File: src/LearnositySdk/Request/DataApi.java [hunk 1]
```diff
@@ -218,7 +218,6 @@ private JSONObject createResponseObject(Remote remote) throws Exception {
     	response.put("body", remote.getBody());
     	response.put("contentType",  remote.getContentType());
     	response.put("statusCode", remote.getStatusCode());
-    	response.put("error", remote.getError());
```
Comment (michaelsharman) [inline src/LearnositySdk/Request/DataApi.java]: @klauste we might want to look at why `getError` was removed, rather than removing this.

Comment (klauste) [reply src/LearnositySdk/Request/DataApi.java]: Just checked it, I removed that so that the exceptions becomes visible to the client. So makes sense.

**Comment by klauste** [comment]
@lud2k thanks for these fixes and especially for adding the maven pom.xml. Makes building much easier.

**Comment by lud2k** [comment]
You are welcome :)!

**Pull request merged**